src/main/java/tc/oc/pgm/api/map/MapInfo.java

```diff
@@ -0,0 +1,113 @@
+package tc.oc.pgm.api.map;
+
+import java.text.Normalizer;
+import java.util.Collection;
+import javax.annotation.Nullable;
+import tc.oc.named.Named;
+import tc.oc.util.Version;
+
+/** Essential information about a map. */
+public interface MapInfo extends Named, Comparable<MapInfo>, Cloneable {
+
+  /**
+   * Get a unique id for the map.
+   *
+   * @return A unique id.
+   */
+  String getId();
+
+  /**
+   * Get the proto of the map's {@link org.jdom2.Document}.
+   *
+   * @see MapProtos
+   * @return The proto.
+   */
+  Version getProto();
+
+  /**
+   * Get the version of the map.
+   *
+   * @return The version.
+   */
+  Version getVersion();
+
+  /**
+   * Get a unique, human-readable name for the map.
+   *
+   * @return A name, alphanumeric with spaces are allowed.
+   */
+  String getName();
+
+  /**
+   * Get a short, human-readable description of the map's objective.
+   *
+   * @return A description.
+   */
+  String getDescription();
```

Comment:
What about a BaseComponent?

Reply:
Components are rendering level stuff, this interface is just descriptive.